Sources/Common/Background Queue/QueueQueryRunner.swift

```diff
@@ -15,18 +16,23 @@ internal class CioQueueQueryRunner: QueueQueryRunner {
         self.logger = logger
     }
 
-    func getNextTask(_ queue: [QueueTaskMetadata], lastFailedTask: QueueTaskMetadata?) -> QueueTaskMetadata? {
-        guard !queue.isEmpty else {
+    func getNextTask(
+        _ inventory: [QueueTaskMetadata],
+        lastRanTask: QueueTaskMetadata?,
+        lastFailedTask: QueueTaskMetadata?
+    ) -> QueueTaskMetadata? {
+        guard !inventory.isEmpty else {
             return nil
         }
+        let inventory = getInventoryTasksLeftToRun(inventory: inventory, lastRanTask: lastRanTask)
```

Comment:
I intentionally modified the original parameter value to avoid human error. 

```swift
fun foo(inventory) {
  let modifiedInventory = ...
  
  // Now in this function, you have 2 "inventories": `inventory` and `modifiedInventory`. 
  // If you use the wrong one by accident, that introduces a bug into this function. 
  //
  // The function body at this time doesn't need 2 versions of inventory. It needs to only use a modified version. 
  // Therefore, if we overwrite the parameter with the modified version, we avoid the possible human error of using the wrong variable from happening. 
```